src/common/JSHandle.ts

```diff
@@ -892,15 +889,16 @@ export class ElementHandle<
     ) => ReturnType | Promise<ReturnType>,
     ...args: SerializableOrJSHandle[]
   ): Promise<WrapElementHandle<ReturnType>> {
-    const defaultHandler = (element: Element, selector: string) =>
-      Array.from(element.querySelectorAll(selector));
     const { updatedSelector, queryHandler } = getQueryHandlerAndSelector(
-      selector,
-      defaultHandler
+      selector
+    );
+    const queryHandlerToArray = Function(
+      'element',
+      'selector',
+      `return Array.from((${queryHandler.queryAll})(element, selector));`
```

Comment:
for my understanding, why does this have to be defined like this and not as an actual function definition?

Reply:
We need some way of inlining `String(queryHandler.queryAll)` into the function body. This PR started with an `eval`-based approach but then we moved towards `Function` which is slightly more minimal.